manifests/charts/istio-cni/README.md

```diff
@@ -21,4 +21,23 @@ helm install istio-cni istio/cni -n kube-system
 ```
 
 Installation in `kube-system` is recommended to ensure the [`system-node-critical`](https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/)
-`priorityClassName` can be used.
+`priorityClassName` can be used. You can install in other namespace only on K8S clusters that allow
+'system-node-critical' outside of kube-system.
+
+## Ambient
+
+To enable ambient, you need to to add `--set cni.ambient.enabled=true`.
+
+## Calico
+
+For Calico, you must also modify the settings to allow source spoofing:
+
+- if deployed by operator,  `kubectl patch felixconfigurations default --type='json' -p='[{"op": "add", "path": "/spec/workloadSourceSpoofing", "value": "Any"}]'`
+- if deployed by manifest, add env `FELIX_WORKLOADSOURCESPOOFING` with value `Any` in `spec.template.spec.containers.env` for daemonset `calico-node`. (This will allow PODs with specified annotation to skip the rpf check. )
+
```

Comment:
I think we may also need  mention adding the IPs as mentioned in this doc?
 https://github.com/istio/istio/tree/master/cni/pkg/ebpf

```kubectl -n istio-system patch ds ztunnel --type='json' -p='[{"op": "add", "path": "/spec/template/metadata/annotations/cni.projectcalico.org~1allowedSourcePrefixes", "value": "[\"10.x.0.0/16\"]"}]'```

Reply:
I don't think patching ztunnel is right - user may reinstall it, upgrade, etc. For debugging or dev it may be ok, not prod.

I added the annotation to ztunnel - so it is added without extra patching ( non-calico will ignore it AFAIK ).

Also instead of 10.x... I set it to 0.0.0.0 - ztunnel may need to impersonate original dst from other clusters in MC or even from ingress.  

Did you do any testing with IPv6 ? Is there an equivalent annotation ?

Reply:
didn't test with IPv6.  I think you are correct about not patching ztunnel.  My comment was that the readme should point out that if a user wants to adjust the allowedSourcePrefix they should make the change to the `daemonset.yaml` Just for my own clarification are you suggesting that users should not set the allowedSourcePrefix?

Reply:
ztunnel is only going to impersonate pods on its own node fwiw, so technically the setting on each ztunnel could be its node's CIDR.

Having a per-pod annotation based on the node is pretty hard to do though so not suggesting we actually do that, but I think it would be the minimal allowed from calico POV

Reply:
Actually I am wrong :-) it will only impersonate *to* or *from* a node-local pod, but it may be `random thing` -> `node local pod`

Reply:
Yes, that was my understanding as well - it should use 'x-forwarded-for' header, which may reflect a source IP from another cluster ( multi-network + E/W ), or the original pod sending request to the Waypoint or even the remote IP for ingress. Since we have no other way to convey this info ( and even when we have metadata endpoint, it's more efficient and compatible to have the real peer IP).

I am starting to believe we should implement HAProxy as well - it is VERY common in cloud infra. How to declare that a pod expects HAProxy remains TBD, maybe something GAMMA should define since others will have the same problem.